src/webview_provider/newLineagePanel.ts

```diff
@@ -264,39 +264,45 @@ export class NewLineagePanel implements LineagePanelView {
     // }
     // this.lruCache.set(node.name, now);
     // const columnsFromDB = this.dbCache.get(node.name)!;
-    const columnsFromDB = await project.getColumnsOfModel(node.name);
-    console.log("addColumnsFromDB: ", node.name, " -> ", columnsFromDB);
-    if (!columnsFromDB || columnsFromDB.length === 0) {
-      return false;
-    }
-    if (columnsFromDB.length > 100) {
-      // Flagging events where more than 100 columns are fetched from db to get a sense of how many of these happen
-      this.telemetry.sendTelemetryEvent(
-        "columnLineageExcessiveColumnsFetchedFromDB",
-      );
-    }
-    const columns: Record<string, ColumnMetaData> = {};
-    Object.entries(node.columns).forEach(([k, v]) => {
-      columns[k.toLowerCase()] = v;
-    });
+    try {
+      const columnsFromDB = await project.getColumnsOfModel(node.name);
+      console.log("addColumnsFromDB: ", node.name, " -> ", columnsFromDB);
+      if (!columnsFromDB || columnsFromDB.length === 0) {
+        return false;
+      }
+      if (columnsFromDB.length > 100) {
+        // Flagging events where more than 100 columns are fetched from db to get a sense of how many of these happen
+        this.telemetry.sendTelemetryEvent(
+          "columnLineageExcessiveColumnsFetchedFromDB",
+        );
+      }
+      const columns: Record<string, ColumnMetaData> = {};
+      Object.entries(node.columns).forEach(([k, v]) => {
+        columns[k.toLowerCase()] = v;
+      });
 
-    columnsFromDB.forEach((c) => {
-      const existing_column = columns[c.column.toLowerCase()];
-      if (existing_column) {
-        existing_column.data_type = existing_column.data_type || c.dtype;
-        return;
+      columnsFromDB.forEach((c) => {
+        const existing_column = columns[c.column.toLowerCase()];
+        if (existing_column) {
+          existing_column.data_type = existing_column.data_type || c.dtype;
+          return;
+        }
+        node.columns[c.column] = {
+          name: c.column,
+          data_type: c.dtype,
+          description: "",
+        };
+      });
+      if (Object.keys(node.columns).length > columnsFromDB.length) {
+        // Flagging events where columns fetched from db are less than the number of columns in the manifest
+        this.telemetry.sendTelemetryEvent("columnLineagePossibleStaleSchema");
       }
-      node.columns[c.column] = {
-        name: c.column,
-        data_type: c.dtype,
-        description: "",
-      };
-    });
-    if (Object.keys(node.columns).length > columnsFromDB.length) {
-      // Flagging events where columns fetched from db are less than the number of columns in the manifest
-      this.telemetry.sendTelemetryEvent("columnLineagePossibleStaleSchema");
+      return true;
+    } catch (exc) {
+      const error = String(exc);
+      window.showErrorMessage("Cannot add columns from DB: " + error);
+      throw new Error(error);
```

Comment:
Why throw again?

Also this doesn't seem the right error boundary as this method is used in loops.